### PR TITLE
Delete DebugSymbols=true setting for the repo build

### PR DESCRIPTION
**Pull request by jkotas**
This property does not do what its name says. The symbols are generated regardless of whether this property is true or false. What this property actually does is that it disables C# peephole IL optimizations.

This change results in ~0.5% IL binary size improvement thanks to the Roslyn IL peephole optimizations that it enables. Also, this gets us closer to default build configuration used by our users.

### CODE REVIEW

**Comment by am11** [comment]
> This property does not do what its name says. The symbols are generated regardless of whether this property is true or false. What this property actually does is that it disables C# peephole IL optimizations.

https://learn.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-properties?view=vs-2022 documents `DebugSymbols` as:
> Setting -p:DebugSymbols=false on the command line disables generation of program database (.pdb) symbol files.

Does it mean documentation is incorrect or it's infact a bug an accidental bug?

**Comment by am11** [comment]
> This property does not do what its name says. The symbols are generated regardless of whether this property is true or false. What this property actually does is that it disables C# peephole IL optimizations.

https://learn.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-properties?view=vs-2022 documents `DebugSymbols` as:
> Setting -p:DebugSymbols=false on the command line disables generation of program database (.pdb) symbol files.

Does it mean documentation is incorrect or it's infact an accidental bug?

**Comment by jkotas** [comment]
> Does it mean documentation is incorrect or it's in fact an accidental bug?

I have opened https://github.com/dotnet/sdk/issues/41001

My conclusion after reading through mentions of `DebugSymbols` property in dotnet/sdk repo is that it is a legacy mess that that should be avoided. `DebugType` property is what actually works.

File: eng/illink.targets [hunk 1]
```diff
@@ -34,7 +34,7 @@
     <EnableTrimAnalyzer Condition="'$(EnableTrimAnalyzer)' == '' And (Exists('$(ILLinkSuppressionsXml)') Or Exists('$(ILLinkSuppressionsConfigurationSpecificXml)'))">false</EnableTrimAnalyzer>
 
     <!-- if building a PDB, tell illink to rewrite the symbols file -->
```
Comment (MichalStrehovsky) [inline eng/illink.targets:36]:
```suggestion
    <!-- tell illink to rewrite the symbols file -->
```

**Review by MichalStrehovsky — APPROVED** [review]
Thanks!

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
For others reviewing this, the Microsoft.NET.Sdk makes this property still default to true when doing a Debug configuration build: https://github.com/dotnet/sdk/blob/ace21a69999eede779921900fca9420d37395aee/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props#L49

**Comment by jkotas** [comment]
Possibly. It depends on whether and how the behavior of `DebugSymbols` is changed: https://github.com/dotnet/msbuild/issues/10183

**Pull request merged**